### PR TITLE
Add consul key/value store deploy-hook

### PR DESCRIPTION
**Pull request by rgynn**
A deployscript that deploys the keyfiles to a consul key/value store http api.

Requires two env variables for example:
DEPLOY_CONSUL_URL="http://localhost:8500/v1/kv"
DEPLOY_CONSUL_ROOT_KEY="acme"

with the -d flag set to: "test.hajhatten_testdomain.com"
```curl -XPUT http://localhost:8500/v1/kv/acme/test.hajhatten_testdomain.com/test.hajhatten_testdomain.com.key --data @/Users/hajhatten/.acme.sh/test.hajhatten_testdomain.com/test.hajhatten_testdomain.com.key```

### CODE REVIEW

File: deploy/consul.sh [hunk 1]
```diff
@@ -0,0 +1,66 @@
+#!/bin/bash
```
Comment (Neilpang) [inline deploy/consul.sh]: use  `sh` shebang

File: deploy/consul.sh [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  _debug DEPLOY_CONSUL_ROOT_KEY "$DEPLOY_CONSUL_ROOT_KEY"
+  
+  # set base url for all uploads
+  upload_base_url="${DEPLOY_CONSUL_URL}/${DEPLOY_CONSUL_ROOT_KEY}/${_cdomain}"
+  _debug upload_base_url "$upload_base_url"
+
+  # private
+  _info uploading "$_ckey"
+  response=$(_post "@${_ckey}" "${upload_base_url}/${_cdomain}.key" "" "PUT")
+  _debug response "$response"
```
Comment (Neilpang) [inline deploy/consul.sh]: `response`  should use `_debug2()`

File: deploy/consul.sh [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  # fullchain
+  _info uploading "$_cfullchain"
+  response=$(_post "@${_cfullchain}" "${upload_base_url}/fullchain.cer" "" "PUT")
+  _debug response "$response"
+
+  return 0
+
+}
+
+####################  Private functions below ##################################
```
Comment (Neilpang) [inline deploy/consul.sh]: add a new line at the end of the file.

**Comment by bgarret** [comment]
PR #3491 has been merged with the same feature, you could close this one?

**Pull request closed without merging**